python/openassetio_traitgen/generators/helpers.py

```diff
@@ -73,9 +73,23 @@ def package_dependencies(
     """
     dependencies = set()
     for declaration in declarations:
-        if isinstance(declaration, SpecificationDeclaration):
-            for trait in declaration.trait_set:
-                dependencies.add(trait.package)
+        dependencies |= set(declaration_dependencies(declaration))
+    dependencies = list(dependencies)
+    dependencies.sort()
+    return dependencies
+
+
+def declaration_dependencies(
+    declaration: Union[SpecificationDeclaration, TraitDeclaration]
+) -> List[str]:
```

Comment:
[Consider] Think these should return sets by default? There's a fair bit of converting going on.

Reply:
They are sorted, so can't really be sets, but yeah we can try to tidy this up a bit.

Reply:
^ see https://github.com/OpenAssetIO/OpenAssetIO-TraitGen/pull/12#discussion_r1137517147